src/main/java/com/zenith/network/server/handler/ProxyServerLoginHandler.java

```diff
@@ -76,35 +76,37 @@ && nonNull(CACHE.getTabListCache().getTabList().get(CACHE.getProfileCache().getP
                     CACHE.getPlayerCache().getPortalCooldown()
             ));
             if (!proxy.isInQueue()) { PlayerCache.sync(); }
+        } else if (clientGameProfile.getId().equals(CACHE.getProfileCache().getProfile().getId())) {
+            connection.disconnect("You can't spectate your own UUID.");
+        } else if (clientGameProfile.getName().equals(CACHE.getProfileCache().getProfile().getName())) {
+            connection.disconnect("You can't spectate your own Username.");
```

Comment:
There's gotta be better solutions than blocking this use case entirely. e.g. change the spectator's game profile at login

Reply:
> There's gotta be better solutions than blocking this use case entirely. e.g. change the spectator's game profile at login

i just wasn't smart enough. :)